File: fastapi/app/models/appointment.py
from sqlalchemy import Column, String, DateTime, Boolean, Integer, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
import uuid
from datetime import datetime
from app.database import Base

class Appointment(Base):
    __tablename__ = "appointments"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    task_id = Column(UUID(as_uuid=True), ForeignKey("tasks.id"), nullable=False)
    date_time = Column(DateTime, nullable=False)
    location = Column(String)
    reminder_mins = Column(Integer)
    is_dday = Column(Boolean, default=False)
    created_at = Column(DateTime, default=datetime.utcnow)

File: app/src/main/java/com/abdessamad/orbyt/data/models/Appointment.kt
package com.abdessamad.orbyt.data.models

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.Date
import java.util.UUID

@Entity(tableName = "appointments")
data class Appointment(
    @PrimaryKey
    val id: UUID = UUID.randomUUID(),
    val taskId: UUID,
    val dateTime: Date,
    val location: String?,
    val reminderMins: Int?,
    val isDday: Boolean = false,
    val createdAt: Date = Date()
)